python/lsst/daf/butler/core/_butlerUri/http.py

```diff
@@ -84,11 +84,22 @@ def getHttpSession() -> requests.Session:
 
     log.debug("Creating new HTTP session...")
 
+    ca_bundle = None
+    try:
+        ca_bundle = os.environ['LSST_BUTLER_WEBDAV_CA_BUNDLE']
+    except KeyError:
+        log.warning("Environment variable LSST_BUTLER_WEBDAV_CA_BUNDLE is not set: "
+                    "HTTPS requests will fail. If you intend to use HTTPS, please "
+                    "export this variable.")
+    session.verify = ca_bundle
+
     try:
         env_auth_method = os.environ['LSST_BUTLER_WEBDAV_AUTH']
     except KeyError:
-        raise KeyError("Environment variable LSST_BUTLER_WEBDAV_AUTH is not set, "
-                       "please use values X509 or TOKEN")
+        log.debug("Environment variable LSST_BUTLER_WEBDAV_AUTH is not set, "
```

Comment:
My problem here is that getHttpSession() currently is not aware of the server URL, so we cannot internally check if the server is a webdav endpoint or not. I could add an optional parameter to getHttpSession() though, but even pure http servers may require x509/token authentication, so maybe it could be useful in some cases, even if the backend is not webdav.
And in that case maybe change the name of the env variable to something like LSST_BUTLER_HTTP_AUTH

Reply:
I think the ability to add new CAs is worthwhile but we shouldn't complain if someone is doing https somewhere on GitHub and uses standard certificates. Now that it's a debug I'm not worried about the log message confusing people when it's not relevant.